internal/validators/network.go

```diff
@@ -32,21 +33,36 @@ type networkRule interface {
 	Name() string
 }
 
-// httpClient defines the interface for the HTTP client used by the PublicBlobRuleService.
-type httpClient interface {
-	Do(req *http.Request) (*http.Response, error)
-}
-
 // NetworkService is a service for network validation.
 type NetworkService struct {
-	httpClient httpClient
+	httpClient *http.Client
 	log        logr.Logger
 }
 
+// NetworkServiceHTTPClientOption allows customizing the NetworkService's HTTP client.
+type NetworkServiceHTTPClientOption func(*http.Client)
+
+// WithTLSConfig allows callers to optionally provide a custom TLS configuration.
+func WithTLSConfig(config *tls.Config) NetworkServiceHTTPClientOption {
+	return func(client *http.Client) {
+		client.Transport = &http.Transport{
+			TLSClientConfig: config,
+		}
+	}
+}
+
 // NewNetworkService creates a new NetworkService.
-func NewNetworkService(httpClient httpClient, log logr.Logger) *NetworkService {
+func NewNetworkService(log logr.Logger, opts ...NetworkServiceHTTPClientOption) *NetworkService {
```

Comment:
This is a great improvement! Happy to approve as-is, but wanted to mention that many libraries I've seen support a `WithTransport` functional option. So the user configures the entire HTTP transport, including TLS config.

What you have is totally fine for the time being though - it's just slightly less flexible.